switch.go

```diff
@@ -313,13 +313,13 @@ func (sw *Switch) DialPeerWithAddress(addr *NetAddress, persistent bool) (*Peer,
 	if persistent {
 		peer.MakePersistent()
 	}
+	sw.dialing.Delete(addr.IP.String())
```

Comment:
fixed in https://github.com/tendermint/go-p2p/pull/18/commits/837b0ed702805815e5cc9ac04599292a222d73db